glfw/build.zig

```diff
@@ -8,14 +8,15 @@ pub fn build(b: *Builder) void {
     const target = b.standardTargetOptions(.{});
 
     const test_step = b.step("test", "Run library tests");
-    test_step.dependOn(&testStep(b, mode, target).step);
+    test_step.dependOn(&testStep(b, mode, target, "glfw_tests", .{}).step);
+    test_step.dependOn(&testStep(b, mode, target, "glfw_shared_tests", .{ .shared = true }).step);
 }
 
-pub fn testStep(b: *Builder, mode: std.builtin.Mode, target: std.zig.CrossTarget) *std.build.RunStep {
-    const main_tests = b.addTestExe("glfw_tests", thisDir() ++ "/src/main.zig");
+pub fn testStep(b: *Builder, mode: std.builtin.Mode, target: std.zig.CrossTarget, exe_name: []const u8, options: Options) *std.build.RunStep {
```

Comment:
Could you leave this function signature as it was before, so that e.g. [this does not break](https://github.com/hexops/mach/blob/e6d7faaabab303d006ea998547c6dc7ce2305c2a/build.zig#L33) - and instead introduce a new private `fn testStepShared` & just duplicate the code here to test the shared library approach?

Reply:
i think this is a good approach and should be applied to all sub-projects

Reply:
not opposed to that, let's do it in another PR though.